src/pycita/pycita.py

```diff
@@ -0,0 +1,696 @@
+        :return: 回执结果. 如果交易还没执行且timeout=0, 则返回{}. 否则表示在pending区块中已经加入此交易, 期待共识
+        """
+        t0 = time.time()
+
+        while 1:
+            r = self._jsonrpc('getTransactionReceipt', [param_to_str(hash_addr)])
+            if r is None:
+                r = {}
+            assert isinstance(r, dict)
+            if timeout == 0 or r:
+                break
+
+            t1 = time.time()
+            if t1 - t0 >= timeout and timeout != -1:
+                raise RuntimeError('timeout')
+            time.sleep(1)
+
+        error = r.get('errorMessage')
+        if error:  # 交易失败
+            raise RuntimeError(error)
+
+        return r
+
+    def call_readonly_func(self, contract_addr: PARAM, func_addr: PARAM, param: PARAM = b'', from_addr: PARAM = b'', block_number: str = 'latest') -> bytes:
+        """
+        调用合约的只读函数.
+
+        :param contract_addr: 合约地址, 20字节
+        :param func_addr: 合约内的函数地址, 4字节
+        :param param: 合约构造函数的参数经encode_param编码后的bytes. 无参数用 b''
+        :param from_addr: 调用者的地址, 默认是 b''
+        :param block_number: 区块高度, 可以是一个'0x'开头的高度字符串, 也可以是'latest'或'pending', 默认是'latest'
+        :return: 返回值编码的bytes
+        """
+        # 构造 CallRequest
+        to_ = param_to_str(contract_addr)
+        assert len(to_) == 40 + 2
+        req = {'to': to_}
+
+        if from_addr:
+            from_ = param_to_str(from_addr)
+            assert len(from_) == 40 + 2
+            req['from'] = from_
+
+        data = param_to_str(func_addr)
+        assert len(data) == 8 + 2
+        if param:
+            data += param_to_str(param)[2:]
+        req['data'] = data
+
+        r = self._jsonrpc('call', [req, block_number])
+        assert isinstance(r, str) and r.startswith('0x')
+        return param_to_bytes(r)
+
+    def call_func(self, private_key: PARAM, contract_addr: PARAM, func_addr: PARAM, param: PARAM = b'', quota: int = DEFAULT_QUOTA) -> str:
+        """
+        调用合约的函数.
+
+        :param private_key: 私钥
+        :param contract_addr: 合约地址
+        :param func_addr: 合约内的函数地址
+        :param param: 编码后的函数参数列表
+        """
+        tx_hash = self.send_transaction(private_key, contract_addr, param_to_bytes(join_param(func_addr, param)), quota=quota)
+        return tx_hash
+
+    def batch_call_func(self, private_key: PARAM, tx_code_list: List[PARAM], quota: int = DEFAULT_QUOTA) -> str:
+        """
+        发起批量交易.
+
+        :param private_key: 私钥.
+        :param tx_code_list: 由ContractClass.get_tx_code生成的交易数据.
+        :return: 交易hash
+        """
+        data: List[bytes] = []
+        for tx_code in tx_code_list:
+            c = param_to_bytes(tx_code)
+            assert len(c) >= 20, 'bad tx_code'
+            head, tail = c[:20], c[20:]
+            n = len(tail)
+            data += [head, n.to_bytes(4, byteorder='big'), tail]
+
+        # 调用 BatchTx 合约的 multiTxs 方法.
+        tx_hash = self.call_func(private_key,
+                                 '0xffffffffffffffffffffffffffffffffff02000e',
```

Comment:
`0xffffffffffffffffffffffffffffffffff02000e ` should be replaced by a constant variable

Reply:
fixed